apps/api/models.py

```diff
@@ -102,24 +105,50 @@ class APIKey(models.Model):
     class Meta:
         db_table = 'api_key'
 
+    def __unicode__(self):
+        return (
+            u'<{cls} user={user}, type={type}, key={key} secret=...>'
+            .format(cls=self.__class__.__name__, key=self.key,
+                    type=self.type, user=self.user))
+
     @classmethod
     def get_jwt_key(cls, **query):
         """
-        Look up the config for a JWT key.
+        return a single APIKey instance for a JWT key matching the query.
         """
+        query.setdefault('is_active', True)
         return cls.objects.get(type=SYMMETRIC_JWT_TYPE, **query)
 
     @classmethod
     def new_jwt_credentials(cls, user, **attributes):
-        # This could get more fancy if we allow multiple keys per user.
-        key = 'user:{}'.format(user.pk)
+        """
+        Generates a new key/secret pair suitable for symmetric JWT signing.
+
+        This method must be run within a db transaction.
+        Returns an instance of APIKey.
+        """
+        key = cls.get_unique_key('u-{}-'.format(user.pk))
```

Comment:
It probably doesn't matter but should we be exposing the user's PK like this? Would it be beneficial to hash it and avoid exposing the PK? I'm not sure if we need it for anything.

I think I liked `user:25` more than `-25-` since we probably wouldn't use `user:` as a prefix if we started keying off something else but we might be tempted to keep using `-{}-` if we switch and have an accidental collision.